crates/threads-xform/src/lib.rs

```diff
@@ -344,9 +344,10 @@ fn inject_start(
             // we give ourselves a stack and we update our stack
             // pointer as the default stack pointer is surely wrong for us.
             |body| {
-                // local = malloc(stack.size) [aka base]
+                // local = malloc(stack.size, align) [aka base]
                 with_temp_stack(body, memory, stack, |body| {
                     body.i32_const(stack.size as i32)
+                        .i32_const(4)
```

Comment:
```suggestion
                        .i32_const(16)
```

I think the stack's actually supposed to be aligned to 16 bytes (source: https://github.com/WebAssembly/tool-conventions/blob/main/BasicCABI.md#the-linear-stack).